src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -636,6 +745,34 @@ public static void AppliedRequestFormLimits(this ILogger logger)
             }
         }
 
+        private static void LogFilterExecutionPlan(ILogger logger, string filterType, IEnumerable<IFilterMetadata> filters)
+        {
+            string[] filterList = null;
+            if (filters != null && filters.Any())
+            {
+                filterList = GetFilterList(filters);
+            }
+
+            _logFilterExecutionPlan(logger, filterType, filterList, null);
```

Comment:
This could probably be simplified by using `Array.Empty` as the sentinel and no accepting null into any of these methods.

`null`-valued `IEnumerable<>` is a trap

Reply:
I noticed that ResourceInvoker throws a ArgumentNull exception if filters are null, so we are probably safe to assume that we wouldn't be in NRE situation? https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs#L60

Reply:
Yeah I don't think filters is ever null here